src/test/kotlin/io/nais/security/oauth2/rsakeystore/KeyStoreTest.kt

```diff
@@ -0,0 +1,54 @@
+package io.nais.security.oauth2.rsakeystore
+
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.nais.security.oauth2.mock.DataSource
+import io.nais.security.oauth2.mock.withMigratedDb
+import org.junit.jupiter.api.Test
+import java.lang.Thread.sleep
+import java.time.LocalDateTime
+
+class KeyStoreTest {
+
+    @Test
+    fun `keyStore should insert new record when db is empty`() {
+        withMigratedDb {
+            with(KeyStore(DataSource.instance)) {
+                val createdRsaKeys = initKeys()
+                val readKeys = read()
+                readKeys shouldNotBe null
+                createdRsaKeys.currentKey shouldBe readKeys?.currentKey
+                createdRsaKeys.previousKey shouldBe readKeys?.previousKey
+                createdRsaKeys.nextKey shouldBe readKeys?.nextKey
+            }
+        }
+    }
+
+    @Test
+    fun `keystore should add new initial record on empty db or update existing record on expired keys`() {
+        withMigratedDb {
+            with(KeyStore(DataSource.instance)) {
+                setExpires(1)
+                val rsaKeysInitial = keys
+                rsaKeysInitial.expired(LocalDateTime.now()) shouldBe false
+                // Go past expiry date
+                sleepy(2)
```

Comment:
no need to slow down the test, can be solved with a configurable TTL for keys and LocalDateTime.plusSeconds()